packages/core/src/common/object.spec.ts

```diff
@@ -0,0 +1,103 @@
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { expect } from 'chai';
+import { deepFreeze, notEmpty, isEmpty, deepClone } from './objects';
+
+describe.only('Object', () => {
+
+    describe('#deepClone', () => {
+
+        describe('primitives', () => {
+            it('should deep clone numbers', () => {
+                expect(1).to.equal(deepClone(1));
+            });
+            it('should deep clone strings', () => {
+                expect('foo').to.equal(deepClone('foo'));
+                expect('').to.equal(deepClone(''));
+            });
+            it('should deep clone booleans', () => {
+                expect(true).to.equal(deepClone(true));
+                expect(false).to.equal(deepClone(false));
+            });
+        });
+
+        describe('undefined', () => {
+            it('should deep clone undefined', () => {
+                expect(undefined).to.equal(deepClone(undefined));
+            });
+        });
+
+        describe('arrays', () => {
+            it('should deep clone arrays', () => {
+                const a = [1, 2, 3];
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(Array.isArray(a)).to.equal(true);
+                expect(Array.isArray(b)).to.equal(true);
+            });
+
+            it('should deep clone nested arrays', () => {
+                const a = [[1], [2]];
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(Array.isArray(a)).to.equal(true);
+                expect(Array.isArray(b)).to.equal(true);
+            });
+        });
+
+        describe('objects', () => {
+            it('should deep clone objects', () => {
+                const a = { 'foo': true, 'bar': 1 };
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(typeof a).to.equal('object');
+                expect(typeof b).to.equal('object');
+            });
+
+            it('should deep clone nested objects', () => {
+                const a = { 'foo': true, 'nested': { 'foo': 1 } };
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(typeof a).to.equal('object');
+                expect(typeof b).to.equal('object');
+            });
+        });
+    });
+
+    describe('#deepFreeze', () => {
+        it('should deep freeze an object', () => {
+            const e = { valeur: 10, boolval: true };
+            const d = deepFreeze(e);
+            expect(true).to.equal(Object.isFrozen(d));
+        });
+    });
+
+    describe('#notEmpty', () => {
+        it('should validate it is not Empty', () => {
+            const z = {};
+            expect(true).to.equal(notEmpty(z));
+        });
```

Comment:
We should test both return types:

```suggestion
        it('should return "true" when not empty', () => {
            expect(notEmpty({})).to.equal(true);
            expect(notEmpty([])).to.equal(true);
            expect(notEmpty({ a: 1 })).to.equal(true);
        });
        it('should return "false" when empty', () => {
            expect(notEmpty(undefined)).to.equal(false);
        });
```

The same is true for `#isEmpty`.